# HG changeset patch
# User Kirill Belokurov <kirill.belokurov@gmail.com>
# Date 1206631632 -7200
# Node ID a0b60931be1cb2eb9b4a4d412f1b6e32ca1c7349
# Parent  8821207817cf34ee0e34c688a4da4c9b8c737529
read only full frames into buffer - otherwise decoder will lose trailing bytes for 24bit AIFF

diff -r 8821207817cf -r a0b60931be1c src/demuxers/demux_aiff.c
--- a/src/demuxers/demux_aiff.c	Wed Mar 26 18:01:20 2008 +0200
+++ b/src/demuxers/demux_aiff.c	Thu Mar 27 17:27:12 2008 +0200
@@ -170,7 +170,9 @@ static int open_aiff_file(demux_aiff_t *
         (this->audio_bits / 8);
       this->running_time = (this->audio_frames / this->audio_sample_rate) * 1000;
 
-      this->audio_block_align = PCM_BLOCK_ALIGN;
+      /* we should send only complete frames to decoder, as it 
+       * doesn't handle underconsumption yet */
+      this->audio_block_align = PCM_BLOCK_ALIGN - PCM_BLOCK_ALIGN % (this->audio_bits / 8 * this->audio_channels);
 
       break;
 
